test/SparqlAntlrParserTest.cpp

```diff
@@ -855,3 +855,33 @@ TEST(SparqlParser, RDFLiteral) {
       "\"1950-01-01T00:00:00\"^^<http://www.w3.org/2001/XMLSchema#dateTime>");
   expectRDFLiteralFails(R"(?a ?b "The \"Moon\""@en .)");
 }
+
+TEST(SparqlParser, SelectQuery) {
+  auto expectSelectQuery = ExpectCompleteParse<&Parser::selectQuery>{
+      {{INTERNAL_PREDICATE_PREFIX_NAME, INTERNAL_PREDICATE_PREFIX_IRI}}};
+  auto expectSelectQueryFails = ExpectParseFails<&Parser::selectQuery>{};
+  expectSelectQuery(
+      "SELECT * WHERE { ?a <bar> ?foo }",
+      m::ParsedQuery("SELECT * WHERE { ?a <bar> ?foo }", m::AsteriskSelect(),
+                     m::GraphPattern(m::Triples({{"?a", "<bar>", "?foo"}}))));
+  expectSelectQuery(
+      "SELECT ?foo WHERE { } GROUP BY ?foo",
+      m::ParsedQuery("SELECT ?foo WHERE { } GROUP BY ?foo",
+                     m::VariablesSelect({"?foo"}), m::GraphPattern(), {}, {},
+                     {Variable{"?foo"}}, {}));
+  expectSelectQuery(
+      "SELECT (COUNT(?foo) as ?baz) WHERE { } GROUP BY ?bar",
+      m::ParsedQuery("SELECT (COUNT(?foo) as ?baz) WHERE { } GROUP BY ?bar",
+                     m::Select({std::pair{"COUNT(?foo)", "?baz"}}),
+                     m::GraphPattern(), {}, {}, {Variable{"?bar"}}));
```

Comment:
I think this query also doesn't work in QLever (all aggregated variables have to be visible in the query), but therefore the Expressions need to return all the variables they contain. 
Add a `TODO` to the parsing function of the SELECT clause/ the checks of the `ParsedQueyr` about this. I will implement such checking soon, because I also need it in other places.

Reply:
Yes adding some way for the an `SparqlExpression` to know all variables visible in its body sounds like a good idea. That should make your proposed constructor nicer than what can be built without this functionality.

Checks should also be added for grouping by a not visible variable and ordering while not grouping by a not visible variable. They both cause errors later on.